ui/src/app/settings/containers/Users/Users.js

```diff
@@ -49,74 +51,64 @@ export const Users = ({ authUser, users }) => {
       "maas-keys": user.sshkeys_count
     }
   }));
-  return (
-    <>
-      <h4>Users</h4>
-      <MainTable
-        defaultSort="username"
-        defaultSortDirection="ascending"
-        headers={[
-          { content: "Username", sortKey: "username" },
-          { content: "Email", sortKey: "email" },
-          {
-            content: "Machines",
-            className: "u-align--right",
-            sortKey: "machines"
-          },
-          { content: "Type", sortKey: "type" },
-          { content: "Last seen", sortKey: "last-seen" },
-          { content: "Role", sortKey: "role" },
-          {
-            content: "MAAS keys",
-            className: "u-align--right",
-            sortKey: "maas-keys"
-          },
-          { content: "Actions", className: "u-align--right" }
-        ]}
-        rows={userRows}
-        sortable={true}
-      />
-    </>
-  );
-};
 
-Users.propTypes = {
-  authUser: UserShape.isRequired,
-  users: PropTypes.arrayOf(UserShape)
-};
+const Users = ({ initialCount = 20 }) => {
+  const [batchSize, setBatch] = useState(initialCount);
+  const dispatch = useDispatch();
+  const users = useSelector(state => selectors.users.get(state, batchSize));
+  const userCount = useSelector(selectors.users.count);
+  const loading = useSelector(selectors.users.loading);
+  const authUser = useSelector(baseSelectors.auth.getAuthUser);
 
-Users.defaultProps = {
-  users: [
-    {
-      email: "admin@example.com",
-      first_name: "",
-      global_permissions: ["machine_create"],
-      id: 1,
-      is_superuser: true,
-      last_name: "",
-      sshkeys_count: 2,
-      username: "admin"
-    },
-    {
-      email: "simon@example.com",
-      first_name: "",
-      global_permissions: ["machine_create"],
-      id: 2,
-      is_superuser: false,
-      last_name: "",
-      sshkeys_count: 9,
-      username: "simon"
+  useEffect(() => {
+    if (!users.length) {
+      dispatch(actions.users.fetch());
     }
-  ]
-};
+  }, [dispatch, users.length]);
```

Comment:
The `users.length` dep means this useEffect runs twice, when you first load the component and then when the users are returned from the API. This would also mean it would run if users are added/removed over the websocket.